lib/hive/controller/android.rb

```diff
@@ -6,13 +6,16 @@ module Hive
   class Controller
     class Android < Controller
       attr_accessor :devices
+      attr_accessor :attached_devices
 
-      def detect
-        all_attached_devices = get_connected_devices # get all connected devices
-        to_poll = select_devices(all_attached_devices) # select devices to poll
+      def detect(device_type = 'Mobile')
+        # device_type should be either 'Mobile' or 'Tv'
+        @device_type = device_type
+        connected_devices = get_connected_devices # get all connected devices
+        to_poll = select_devices(connected_devices) # select devices to poll
         poll_devices(to_poll) # poll devices
         register_new_devices(devices) # register new devices with Hivemind
```

Comment:
Where is devices set? If it's using `@devices` (which it appears to be doing) then do we need the parameter for the register_new_devices method?